students/charlie_england/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,122 @@
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320)
+    status = peewee.CharField(max_length=10)
+    credit_limit = peewee.FloatField()
+
+database.create_tables([Customer])
+
+def add_customer(customer_id, name, lastname, address, phone_number, email, status, credit_limit):
+    """
+        adds a new customer to the database.
+    """
+    try:
+        with database.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+            )
+            customer.save()
+            logging.info(f'Customer {customer_id}/{name} created and added to database')
+    except Exception as e:
+        logging.error(f'Could not add {customer_id} due to {e}')
+
+def search_customer(customer_id):
+    """
+        searches for an active customer given a customer ID
+        returns a dictionary of the name, lastname, phone number and email
+        if customerID not found, returns an empty dictionary
+    """
+    try:
+        customer = Customer.get(Customer.customer_id == customer_id)
+        logging.info(f'Customer {customer_id} found, returning information')
+        return {
+            'name':customer.name,
+            'lastname':customer.lastname,
+            'phone_number':customer.phone_number,
+            'email':customer.email,
+        }
+    except Exception as e:
+        logging.info(f'Customer {customer_id} not found with error:{e}')
+        return {}
+
+
+
+def delete_customer(customer_id):
+    """
+        deletes the customer from the database
+    """
+    try:
+        with database.transaction():
+            cust = Customer.get_by_id(customer_id)
+            cust.delete_instance()
+        logging.info(f'Customer {customer_id} has been successfully deleted')
+        return True
+    except Exception as e:
+        logging.error(f'Customer {customer_id} could not be deleted, error: {e}')
+
+def update_customer_credit(customer_id, credit_limit):
+    """
+        given a customer id, updates the credit limit
+    """
+    try:
+        with database.transaction():
+            cust = Customer.get_by_id(customer_id)
+            cust.credit_limit = credit_limit
+            cust.save()
+        logging.info(f'Customer {customer_id} has been updated with a new {credit_limit} credit limit')
+    except Exception as e:
+        logging.error(f'Could not update credit limit for {customer_id} due to error: {e}')
+        raise ValueError
+
+def list_active_customers():
+    """
+        will return an integer with the number of active customers
+        active customers have a status of active
+    """
+    try:
+        count = 0
+        for _ in Customer.select().where(Customer.status == 'active'):
```

Comment:
A neater way to do this would be:

num_customers = len(Customer.select().where(Customer.status == 'active'))
logging.info(f'Collected information on active customers, currently {num_customers} active customers')
return num_customers

This way you don't need to iterate through each customer to count 1